src/System.ComponentModel.Composition/src/Microsoft/Internal/GenerationServices.cs

```diff
@@ -71,10 +72,7 @@ public static ILGenerator CreateGeneratorForPublicConstructor(this TypeBuilder t
         /// <returns></returns>
         public static void LoadValue(this ILGenerator ilGenerator, object value)
         {
-            if(ilGenerator == null)
-            {
-                throw new ArgumentNullException(nameof(ilGenerator));
-            }
+            Debug.Assert(ilGenerator == null);
```

Comment:
it should be the other way around. Assert is same as Assumes or Requires. It asserts (checks) that something is true. It's only evaluated during debug builds/tests, that's why it's better as it is removed from the release build and therefore the code path is faster then without the unnecessary check.